### PR TITLE
go_1_17: 1.17.9 -> 1.17.10

### PR DESCRIPTION
**Pull request by otavio**
- go: 1.17.9 -> 1.17.10
- go: 1.18.1 -> 1.18.2

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/compilers/go/1.17.nix [hunk 1]
```diff
@@ -167,7 +167,6 @@ stdenv.mkDerivation rec {
     ./remove-test-pie-1.15.patch
     ./creds-test.patch
     ./go-1.9-skip-flaky-19608.patch
-    ./go-1.9-skip-flaky-20072.patch
```
Comment (zowoq) [inline pkgs/development/compilers/go/1.17.nix:170]: Committed by accident?

**Comment by zowoq** [comment]
1.18 can go to master, please move it to a separate PR.

**Comment by zowoq** [comment]
Commit titles should be `go_1_17` `go_1_18` instead of just `go` for `ofborg` to built them automatically.

File: pkgs/development/compilers/go/1.17.nix (hunk 1 above)
Comment (otavio) [reply pkgs/development/compilers/go/1.17.nix:170]: No. It fails to build.

**Review by SuperSandro2000 — APPROVED** [review]
@zowoq

**Pull request merged**